commerce-test/src/testIntegration/java/com/liferay/commerce/shipment/test/CommerceShipmentTest.java

```diff
@@ -0,0 +1,1024 @@
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
+
+import java.math.BigDecimal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.frutilla.FrutillaRule;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Riccardo Alberti
+ */
+@RunWith(Arquillian.class)
+public class CommerceShipmentTest {
+
+	@ClassRule
+	@Rule
+	public static AggregateTestRule aggregateTestRule = new AggregateTestRule(
+		new LiferayIntegrationTestRule(), PermissionCheckerTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_company = CompanyTestUtil.addCompany();
+
+		_user = UserTestUtil.addUser(_company);
+
+		_commerceCurrency = CommerceCurrencyTestUtil.addCommerceCurrency(
+			_company.getGroupId());
+
+		_commerceOrders = new ArrayList<>();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		for (CommerceOrder commerceOrder : _commerceOrders) {
+			_commerceOrderLocalService.deleteCommerceOrder(commerceOrder);
+		}
+
+		_companyLocalService.deleteCompany(_company);
+	}
+
+	@Test(expected = CommerceOrderShippingAddressException.class)
+	public void testCheckoutWihoutShippingAddress() throws Exception {
+		frutillaRule.scenario(
+			"It should not be possible to create an order without shipping " +
+				"address"
+		).given(
+			"An order is created without a shipping address"
+		).when(
+			"Checkout is executed"
+		).then(
+			"An exception is raised"
+		);
+
+		CommerceChannel commerceChannel = _createCommerceChannel();
+
+		BigDecimal value = BigDecimal.valueOf(RandomTestUtil.nextDouble());
+
+		CommerceOrder commerceOrder = CommerceTestUtil.addB2CCommerceOrder(
+			_company.getGroupId(), _user.getUserId(),
+			_commerceCurrency.getCommerceCurrencyId(),
+			commerceChannel.getSiteGroupId());
+
+		CommerceShippingMethod commerceShippingMethod =
+			CommerceTestUtil.addFixedRateCommerceShippingMethod(
+				commerceOrder.getGroupId(), value);
+
+		commerceOrder.setCommerceShippingMethodId(
+			commerceShippingMethod.getCommerceShippingMethodId());
+
+		_commerceOrderLocalService.updateCommerceOrder(commerceOrder);
+
+		_commerceOrders.add(commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			commerceOrder.getCommerceCurrency(), commerceChannel, null, null,
+			null, commerceOrder);
+
+		_commerceOrderLocalService.checkoutCommerceOrder(
+			commerceOrder.getCommerceOrderId(), commerceContext,
+			ServiceContextTestUtil.getServiceContext(
+				commerceOrder.getGroupId()));
+	}
+
+	@Test
+	public void testCreateMultipleShippingForSameOrder() throws Exception {
+		frutillaRule.scenario(
+			"An order with multiple shippings is created"
```

Comment:
Throughout this class `shipping` / `shippings` is used as a noun -- instead, it should be `shipment` / `shipments`.